### PR TITLE
feat: add negative test flag to the tests

### PR DESCRIPTION
**Pull request by nicufk**
## Pull request description 
Closes https://github.com/kubeshop/testkube/issues/2482

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by jfermi — APPROVED** [review]

File: cmd/kubectl-testkube/commands/tests/create.go [hunk 1]
```diff
@@ -159,6 +160,7 @@ func NewCreateTestsCmd() *cobra.Command {
 	cmd.Flags().StringVar(&jobTemplate, "job-template", "", "job template file path for extensions to job template")
 	cmd.Flags().StringVarP(&preRunScript, "prerun-script", "", "", "path to script to be run before test execution")
 	cmd.Flags().StringVar(&scraperTemplate, "scraper-template", "", "scraper template file path for extensions to scraper template")
+	cmd.Flags().BoolVar(&negativeTest, "negative-test", false, "negative test if set to true will succeed if test fails and fail if test succeeds")
```
Comment (tkonieczny) [inline cmd/kubectl-testkube/commands/tests/create.go:163]: Maybe something like this?
```
Negative test, if enabled, makes failure an expected and correct test result. If the test fails the result will be set to success, and vice versa. ```

File: cmd/kubectl-testkube/commands/tests/update.go [hunk 2]
```diff
@@ -115,6 +116,7 @@ func NewUpdateTestsCmd() *cobra.Command {
 	cmd.Flags().StringVar(&jobTemplate, "job-template", "", "job template file path for extensions to job template")
 	cmd.Flags().StringVarP(&preRunScript, "prerun-script", "", "", "path to script to be run before test execution")
 	cmd.Flags().StringVar(&scraperTemplate, "scraper-template", "", "scraper template file path for extensions to scraper template")
+	cmd.Flags().BoolVar(&negativeTest, "negative-test", false, "negative test if set to true will succeed if test fails and fail if test succeeds")
```
Comment (tkonieczny) [inline cmd/kubectl-testkube/commands/tests/update.go:119]: `Negative test, if enabled, makes failure an expected and correct test result. If the test fails the result will be set to success, and vice versa. `?

File: pkg/executor/client/job.go [hunk 3]
```diff
@@ -330,6 +330,20 @@ func (c JobExecutor) stopExecution(ctx context.Context, l *zap.SugaredLogger, ex
 	}
 
 	execution.Stop()
+	if isNegativeTest {
+		if result.IsFailed() {
```
Comment (vsukhin) [inline pkg/executor/client/job.go:334]: Should we checked only for failed or aborted also?

**Review by vsukhin — APPROVED** [review]
looks good, one question

**Review by tkonieczny — APPROVED** [review]

**Pull request merged**